include/deal.II/lac/arpack_solver.h

```diff
@@ -482,9 +497,13 @@ void ArpackSolver::solve (
           Assert (false, ExcArpackInfodneupd(info));
         }
 
-      for (size_type i=0; i<eigenvectors.size(); ++i)
+      const unsigned int n_eigenvecs = eigenvectors.size() / 2;
+      for (size_type i=0; i<n_eigenvecs; ++i)
         for (unsigned int j=0; j<n; ++j)
-          eigenvectors[i](j) = v[i*n+j];
+          {
+            eigenvectors[i](j) = z[2*i*n+j];
+            eigenvectors[i+n_eigenvecs](j) = z[(2*i+1)*n+j];
```

Comment:
I agree and will do in part.

Currently, I neither understand Arpack sufficiently for doing a complete make-over, nor do I understand the design decisions that entered the wrapper. I will see if I can persuade a student to look into it for credit.